### ISSUE TITLE
Allow passing a fixed set of base_samples into sample

### ISSUE THREAD

**Issue by Balandat** [cornellius-gp/gpytorch#239]
In certain situations we'd like to be able to fix the base samples in the `zero_mean_mvn_samples` functions. One use case for this is using MC-based batch acquisition functions for Bayesian Optimization, in which a large number of samples are drawn from the posterior to MC-approximate the integral involved in the acquisition function. This sampling results in stochastic function and gradient evaluations, which we can't use directly with non-stochastic (quasi-) higher order optimization algorithms. Now if we fix the underlying base_samples throughout the different iterations of the optimization, then we can use these optimizers (the result will be biased, but that may be good enough or at least a good starting point for other optimization).

To simplify this, I suggest we add a version of `sample` to the random variables that allows to pass in a set of base samples (i.i.d. N(0,1)). 

@jacobrgardner, @gpleiss any thoughts on this?

cc @darbour, @bletham, @bkarrer

**Comment by Balandat** [comment]
I have a draft for this, about to put up a PR.

> Do they have an interface for this already?

@jacobrgardner What do you mean exactly? An interface for torch.distributions? Or for passing in base samples for a MVN?

**Comment by jacobrgardner** [comment]
Yeah I meant for passing in a base samples. Just checking if whatever we do for this now will still be possible when we are constrained to the `torch.distributions` interface

**Comment by Balandat** [comment]
So I don't see an api for passing in base_samples into distributions from `torch.distribution`, but it wouldn't be hard at all to add this in a subclassed version (we'll want to subclass all of the rvs anyway so things work with LazyTensors). Let me start prototyping random_variables that use `torch.distributions` under the hood.
